src/tiledimage.js

```diff
@@ -1850,6 +1875,25 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
     },
 
     /**
+     * @private
+     * @inner
+     * Try to find existing cache of the tile
+     * @param {OpenSeadragon.Tile} tile
+     */
+    _tryFindTileCacheRecord: function(tile) {
+        let record = this._tileCache.getCacheRecord(tile.originalCacheKey);
+
+        if (!record) {
+            return false;
+        }
+        tile.loading = true;
+        this._setTileLoaded(tile, record.data, null, null, record.type);
```

Comment:
It seems like there might be something circular or at least duplicated here. `_setTileLoaded` calls `tile.addCache` which creates a new cache record, but we already found this tile's data in the cache. How does this get sorted out?

Reply:
setTileLoaded finds out the tile data is already present, and makes only sure that:
 - it receives a reference to same caches as any other tile that is already part of the system (namely original cache and main cache if it differs)
 - it fires `tile-loaded` event only

Reply:
Ok, I think I was able to follow the logic through and see how this happens. Thanks for the explanation.